Various improvements and small fixes of the MET uncertainty tools

mmarionncern opened the pull request:
This PR contains several fixes, improvements  and simplification of the MET uncertainty code and MET correction code. Also prepare the handling of the MVA MET uncertainties, planned for 76X.

Modified packages are :
- DataFormats/PatCandidates
- JetMETCorrections/Type1MET
- PhysicsTools/PatAlgos
- PhysicsTools/PatUtils

Code modifications :
- add the photon ES uncertainties in the pat::MET format, was missing before
- add automatic swich for JEC corrections betwee data and MC in the MET type1 code
- move to CHS jets and CHS JEC correctiosn for the type1 MET per default
- jet soft cleaning (electron/muon overlap removal) added per default in the uncertainty tool
- update of the XY MET shift correction (74X version, latest one)
- fix a bug in the XY MET shift correction computation
- update of the type0 correction (74X version, latest one)
- add shortcuts functions for the miniAOD production and re-production, with the minimal set of options needed by non-expert people. 
- add example config file to show how to reprocess the MET on top of miniAODs

Expected changes :
- central MET value will change a bit due to the switch to CHS, no strong effect expected
- JES/JER MET uncertainties can change a bit due to the jet cleaning
- XY and T0 changes will affect all METs with at least one of those corrections

Commit:
update MET correction tool: PhotonES, automatic function, preparatino for MVA MET

Commit:
cleaning + jet cleansing set to electrons/muons only

Commit:
move to CHS T1 at the RECO level

Commit:
update the MET XY correction and met reprocessing example

Commit:
set the automatic switch of JECs for MC and data, tags can still be specified by the user

cmsbuild commented:
A new Pull Request was created by @mmarionncern  for CMSSW_7_5_X.

Various improvements and small fixes of the MET uncertainty tools

It involves the following packages:

DataFormats/PatCandidates
JetMETCorrections/Type1MET
PhysicsTools/PatAlgos
PhysicsTools/PatUtils


@cmsbuild, @vadler, @monttj can you please review it and eventually sign? Thanks.
@rappoccio, @imarches, @ahinzmann, @acaudron, @TaiSakuma, @jdolen, @nhanvtran, @schoef, @ferencek, @gpetruc, @mariadalfonso, @pvmulder this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

Commit:
add GenMETExtractor for MET reprocessing in miniAOD

Commit:
fix data switch and CHS handling

monttj commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #10240 was updated. @cmsbuild, @vadler, @monttj can you please check and sign again.

cmsbuild commented:
-1 
 
Tested at: cf079fd7f37f3ac91cf55ed59336ef2dc6f8c630
I found errors in the following unit tests: 
 
 ---> test testJetMETCorrectionsType1MET had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/2837/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/2837/summary.html

monttj commented:
@mmarionncern 
The test failed with corrMET_pfMet_cfg.py file. The error message is below.  

An exception of category 'ProductNotFound' occurred while
   [0] Processing run: 1 lumi: 2 event: 101
   [1] Running path 'p'
   [2] Calling event method for module PFJetMETcorrInputProducer/'corrPfMetType1'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: reco::JetCorrector
Looking for module label: ak4PFCHSL1FastL2L3Corrector
Looking for productInstanceName:

mmarionncern commented:
@monttj it is on my to do list, will be fixed tomorrow afternoon at the latest

Commit:
fix reco MET type1 JES correctors

cmsbuild commented:
Pull request #10240 was updated. @cmsbuild, @vadler, @monttj can you please check and sign again.

monttj commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: f41696003aba8ab531873fc73ea3572ab3c5c23b
I found errors in the following unit tests: 
 
 ---> test testJetMETCorrectionsType1MET had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/5268/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/5268/summary.html

There are some workflows for which there are errors in the baseline:
 50202.0 step 2
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

mmarionncern commented:
Looks like a DAS error, not sure if it comes from that PR of from a temporary issue with DAS

schoef committed:
Added Alexx Summer15_50nsV2_DATA .db file from Aug. 9th

schoef committed:
adding L2L3Residual level in PhysicsTools/PatAlgos/plugins/JetCorrFactorsProducer.cc

mariadalfonso commented:
Hi all, what is the status of this PR ? Maria

Commit:
update postfix and residuals uncertainties

Commit:
update JEC and uncertainties

Commit:
fix typo

Commit:
move to 25ns XY MET corrections

mmarionncern commented:
@cmsbuild, @vadler, @monttj could you please review the PR ?

cmsbuild commented:
Pull request #10240 was updated. @cmsbuild, @vadler, @monttj can you please check and sign again.

Commit:
fix MET DQM

Conflicts:
	DQMOffline/JetMET/python/jetMETDQMOfflineSourceMC_cff.py
	DQMOffline/JetMET/python/jetMETDQMOfflineSource_cff.py

Commit:
fix MET DQM...

Conflicts:
	DQMOffline/JetMET/python/jetMETDQMOfflineSource_cff.py

Commit:
fix small late user-case typos

monttj commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #10240 was updated. @cmsbuild, @vadler, @danduggan, @deguio, @monttj can you please check and sign again.

cmsbuild commented:
-1
@mmarionncern This pull request cannot be automatically  merged, could you please rebase it?
 you can see the log for git cms-merge-topic here: 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/7517/git-merge-result

mmarionncern commented:
@cmsbuild branch has been rebased

cmsbuild commented:
Pull request #10240 was updated. @cmsbuild, @vadler, @danduggan, @deguio, @monttj can you please check and sign again.

mmarionncern commented:
@cmsbuild @monttj : can the tests be triggered? Thanks

smuzaffar commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: 0fcaeb7696b98918aa94d7c4427f965c9bca272c
 When I ran the RelVals I found an error in the following worklfows: 
 4.22 step1
<pre>DAS Error</pre>
1000.0 step1
<pre>DAS Error</pre>
1001.0 step1
<pre>DAS Error</pre>
1003.0 step1
<pre>DAS Error</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/7562/summary.html

gpetruc commented:
Hi,

According to DAS, all the needed files for these Matrix tests for these
runs are not at cern, i.e. it's not a glitch of the system and re-doing
"please test" won't help.

1003.0 uses `dataset=/MinimumBias/Run2012A-v1/RAW run=191226` whose
corresponding block
`/MinimumBias/Run2012A-v1/RAW#b0847904-85d5-11e1-a9bb-003048caaace` is only
at T0_CH_CERN and T1_US_FNAL (
https://cmsweb.cern.ch/das/request?input=site%20block%3D/MinimumBias/Run2012A-v1/RAW%23b0847904-85d5-11e1-a9bb-003048caaace&instance=prod/global&idx=0&limit=10
)

4.22 uses `/Cosmics/Run2011A-v1/RAW run=160960` whose block
/Cosmics/Run2011A-v1/RAW#d299346e-5384-11e0-9b51-00151755cb60
<https://cmsweb.cern.ch/das/request?input=block%3D%2FCosmics%2FRun2011A-v1%2FRAW%23d299346e-5384-11e0-9b51-00151755cb60&instance=prod%2Fglobal>
is at T0, CNAF and FNAL

1000.0, 1001.0 use `/MinimumBias/Run2011A-v1/RAW run=165121` whose block is
`/MinimumBias/Run2011A-v1/RAW#f6f3c7cc-7fbf-11e0-b519-003048caaace`, at T0,
FNAL, GRIF

I have made a Phedex request to bring them in the relval pool (
https://cmsweb.cern.ch/phedex/prod/Request::View?request=471555 )

Giovanni

On Mon, Aug 24, 2015 at 11:37 AM, cmsbuild <notifications@github.com> wrote:

> -1
> Tested at: 0fcaeb7
> <https://github.com/cms-sw/cmssw/commit/0fcaeb7696b98918aa94d7c4427f965c9bca272c>
> When I ran the RelVals I found an error in the following worklfows:
> 4.22 step1
>
> DAS Error
>
> 1000.0 step1
>
> DAS Error
>
> 1001.0 step1
>
> DAS Error
>
> 1003.0 step1
>
> DAS Error
>
> you can see the results of the tests here:
>
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/7562/summary.html
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/10240#issuecomment-134112060>.
>

gpetruc commented:
samples have been transferred, can somebody restart the tests?

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 0fcaeb7696b98918aa94d7c4427f965c9bca272c
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/7604/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/7604/summary.html

gpetruc commented:
ping?

monttj commented:
+1

gpetruc commented:
@danduggan @deguio : this requires also a DQM signature (if it didn't at the beginning when it was created, so maybe you had missed it)

Commit:
fix nultithread producers and class version

Commit:
fix thread safety, and condition reading from DB

Conflicts:
	JetMETCorrections/Type1MET/plugins/MultShiftMETcorrInputProducer.cc
	JetMETCorrections/Type1MET/plugins/MultShiftMETcorrInputProducer.h

Commit:
last fix on the XY shift correcton producer

Conflicts:
	JetMETCorrections/Type1MET/plugins/MultShiftMETcorrInputProducer.cc

deguio commented on DQMOffline/JetMET/python/jetMETDQMOfflineSource_cff.py:
```diff
@@ -61,6 +67,11 @@
     dqmAk4PFCHSL1FastL2L3ResidualCorrector
 )
 
+dqmAk4PFCHSL1FastL2L3Corrector = ak4PFCHSL1FastL2L3Corrector.clone()
```
ciao @gpetruc 
aren't these lines duplicated? see line 50

arizzi replied:
@mmarionncern  can you please comment?

mmarionncern replied:
@deguio @arizzi  Those lines are complementary. One contains the residuals JEC and the other not.

arizzi commented:
@deguio can you please sign then?

cmsbuild commented:
Pull request #10240 was updated. @cmsbuild, @vadler, @danduggan, @deguio, @monttj can you please check and sign again.

monttj commented:
+1

deguio commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_5_X IBs once checked with relvals in the development release cycle of CMSSW or unless it breaks tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_5_X IBs once checked with relvals in the development release cycle of CMSSW or unless it breaks tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: aa33acdc4ac77509353a29c09e959d4202a3b3c5
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/8087/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_5_X IBs once checked with relvals in the development release cycle of CMSSW (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10240/8087/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_5_X IBs once checked with relvals in the development release cycle of CMSSW (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

davidlange6 commented:
+1
